archimedes/src/app/app.component.html

```diff
@@ -1,36 +1,58 @@
-            (newsResults)="updateNewsResults($event)"
-            (newsLoading)="newsLoading = $event"
-            (newsError)="newsError = $event">
-        </app-search-box>
-
-        <p *ngIf="programLoading">Loading programs...</p>
-        <p *ngIf="programError">Error loading programs</p>
-
-        <p *ngIf="newsLoading">Loading news articles...</p>
-        <p *ngIf="newsError">Error loading news articles</p>
-
-        <div class="program-results" *ngIf="programResults">
-            <h2>Programs</h2>
-            <p>{{programResults.count}} programs found</p>
-            <ul>
-                <app-program *ngFor="let program of programResults.results | slice:0:5" [program]="program"></app-program>
-            </ul>
+        <div class="col mt-5">
+            <app-search-box (programResults)="updateProgramResults($event)" (programLoading)="programLoading = $event"
+                (programError)="programError = $event" (newsResults)="updateNewsResults($event)"
+                (newsLoading)="newsLoading = $event" (newsError)="newsError = $event">
+            </app-search-box>
+
+
+        </div>
+    </div>
+    <div class="row mt-4">
+        <div class="col-lg-6">
+
+            <div class="program-results" *ngIf="programResults || programLoading || programError">
+                <span class="badge badge-default badge-pill float-right mt-1"
+                    *ngIf="programResults && programResults.count">{{programResults.count}}</span>
+                <h2 class="h4 text-uppercase d-line-block mb-3">
+                    Programs <app-loading-icon *ngIf="programLoading" (programLoading)="programLoading = $event">
+                    </app-loading-icon>
+                </h2>
+
+                <div class="alert alert-danger" *ngIf="programError">
+                    Error loading programs
+                </div>
+                <div class="alert alert-info" *ngIf="programResults && programResults.results.length === 0">
+                    No Programs Found
+                </div>
+
+                <ng-container *ngIf="programResults && programResults.results">
+                    <div app-programs class="list-group" [programs]="programResults.results"></div>
+                </ng-container>
+            </div>
+
         </div>
+        <div class="col-lg-6">
+
+            <div class="news-results mt-4 mt-lg-0" *ngIf="newsResults || newsLoading || newsError">
+                <h2 class="h4 text-uppercase mb-3">
+                    News <app-loading-icon *ngIf="newsLoading" (newsLoading)="newsLoading = $event">
+                    </app-loading-icon>
+                </h2>
+
+                <div class="alert alert-danger" *ngIf="newsError">
+                    Error loading news articles
+                </div>
+                <div class="alert alert-info" *ngIf="newsResults && newsResults.length === 0">
+                    No News Articles Found
+                </div>
+                <div class="container">
```

Comment:
You could avoid nesting `.container`s by restructuring the rendered HTML to something like:

```html
<div class="news-results ...">
...
  <div class="list-group">
    <a class="list-group-item">
      <div class="row">
          <div class="col-4">...</div>
          <div class="col-8">...</div>
      </div>
    </a>
  </div>
...
</div>
```